coordinator/src/main/scala/filodb.coordinator/queryplanner/DefaultPlanner.scala

```diff
@@ -525,4 +642,77 @@ object PlannerUtil extends StrictLogging {
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ApplyInstantFunctionRaw =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[RawSeries])
+      case lp: Aggregate =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: BinaryJoin =>
+        lp.copy(lhs = rewritePlanWithRemoteRawExport(lp.lhs, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan],
+          rhs = rewritePlanWithRemoteRawExport(lp.rhs, rangeSelector, additionalLookback)
+            .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ScalarVectorBinaryOperation =>
+        lp.copy(vector = rewritePlanWithRemoteRawExport(lp.vector, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ApplyMiscellaneousFunction =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ApplySortFunction =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ScalarVaryingDoublePlan =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ScalarTimeBasedPlan => lp
+      case lp: VectorPlan =>
+        lp.copy(scalars = rewritePlanWithRemoteRawExport(lp.scalars, rangeSelector, additionalLookback)
+          .asInstanceOf[ScalarPlan])
+      case lp: ScalarFixedDoublePlan => lp
+      case lp: ApplyAbsentFunction =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ApplyLimitFunction =>
+        lp.copy(vectors = rewritePlanWithRemoteRawExport(lp.vectors, rangeSelector, additionalLookback)
+          .asInstanceOf[PeriodicSeriesPlan])
+      case lp: ScalarBinaryOperation => lp
+      case lp: SubqueryWithWindowing =>
+        lp.copy(innerPeriodicSeries =
+          rewritePlanWithRemoteRawExport(lp.innerPeriodicSeries, rangeSelector, additionalLookback)
+            .asInstanceOf[PeriodicSeriesPlan])
+      case lp: TopLevelSubquery =>
+        lp.copy(innerPeriodicSeries =
+          rewritePlanWithRemoteRawExport(lp.innerPeriodicSeries, rangeSelector, additionalLookback = additionalLookback)
+            .asInstanceOf[PeriodicSeriesPlan])
+      case lp: RawSeries =>
+        // IMPORTANT: When we export raw data over remote data call, offset does not mean anything, instead
+        // do a raw lookback of original lookback + offset and set offset to 0
```

Comment:
Why a warning for non zero offset?

Reply:
> When we export raw data over remote data call, offset does not mean anything

What does that mean? In general, plans like `foo{} offset 1h` executed remotely will not account for the offset?

Reply:
Offset will affect the start and end time of the underlying raw data read, thats why at the leaf level where the raw data is read as long as the start and end range of the data is altered based on offset, it means nothing.  Thus instead of reading from 10:00 - 10:05 with no offset we will read 9:00- 9:05 with offset of 1h.

Reply:
Ah I see, I've misunderstood the problem as some bug in the RemoteExec API -- marking resolved